apps/web/pages/api/v1/environments/[environmentId]/surveys/[surveyId]/duplicate/[targetEnvironmentId].ts

```diff
@@ -0,0 +1,159 @@
+  }
+
+  const hasAccess = await hasEnvironmentAccess(req, res, environmentId);
+  const hasTargetEnvAccess = await hasEnvironmentAccess(req, res, targetEnvironmentId);
+
+  if (!hasAccess || !hasTargetEnvAccess) {
+    return res.status(403).json({ message: "Not authorized" });
+  }
+
+  // POST
+  else if (req.method === "POST") {
+    // duplicate current survey including its triggers
+    const existingSurvey = await prisma.survey.findFirst({
+      where: {
+        id: surveyId,
+        environmentId,
+      },
+      include: {
+        triggers: {
+          include: {
+            eventClass: true,
+          },
+        },
+        attributeFilters:{
+          include:{
+            attributeClass:true
+          }
+        }
+      },
+    });
+
+    if (!existingSurvey) {
+      return res.status(404).json({ message: "Survey not found" });
+    }
+
+    let targetEnvironmentTriggers: string[] = [];
+    // map the local triggers to the target environment
+    for (let i = 0; i < existingSurvey.triggers.length; i++) {
+      const trigger = existingSurvey.triggers[i];
+      const targetEnvironmentTrigger = await prisma.eventClass.findFirst({
+        where: {
+          name: trigger.eventClass.name,
+          environment: {
+            id: targetEnvironmentId,
+          },
+        },
+      });
+      if (!targetEnvironmentTrigger) {
+        // if the trigger does not exist in the target environment, create it
+        const newTrigger = await prisma.eventClass.create({
+          data: {
+            name: trigger.eventClass.name,
+            environment: {
+              connect: {
+                id: targetEnvironmentId,
+              },
+            },
+            createdAt: new Date(),
+            updatedAt: new Date(),
+            description: trigger.eventClass.description,
+            type: trigger.eventClass.type,
+            noCodeConfig: trigger.eventClass.noCodeConfig
+              ? JSON.parse(JSON.stringify(trigger.eventClass.noCodeConfig))
+              : undefined,
+          },
+        });
+        targetEnvironmentTriggers.push(newTrigger.id);
+      } else {
+        targetEnvironmentTriggers.push(targetEnvironmentTrigger.id);
+      }
+    }
+
+
+    let targetEnvironmentAttributeFilters: string[] = [];
+    // map the local attributeFilters to the target env
+    for (let i = 0; i < existingSurvey.attributeFilters.length; i++) {
+      const attributeFilter = existingSurvey.attributeFilters[i];
+      // check if attributeClass exists in target env.
+      // if not, create it
+      const targetEnvironmentAttributeClass = await prisma.attributeClass.findFirst({
+        where: {
+          name: attributeFilter.attributeClass.name,
+          environment: {
+            id: targetEnvironmentId,
+          },
+        },
+      });
+      if (!targetEnvironmentAttributeClass) {
+        const newAttributeClass=await prisma.attributeClass.create({
+          data:{
```

Comment:
It looks like your code isn't automatically run through prettier. We use prettier to always format the code automatically in the same style. You can find prettier for VSCode here: https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode
It's also recommended to use it as default formatter and format on save so every time you save a file, it automatically formats the file according to the project rules 😊